sql/tsql/TSqlParser.g4

```diff
@@ -626,6 +626,69 @@ class_type_for_parallel_dw
     | OBJECT
     ;
 
+// https://docs.microsoft.com/en-us/sql/t-sql/statements/grant-transact-sql?view=sql-server-ver15
+// SELECT DISTINCT '| ' + CLASS_DESC
+// FROM sys.dm_audit_actions
+// ORDER BY 1
+class_type_for_grant
+    : APPLICATION ROLE
+    | ASSEMBLY
+    | ASYMMETRIC KEY
+    | AUDIT
+    | AVAILABILITY GROUP
+    | BROKER PRIORITY
+    | CERTIFICATE
+    | COLUMN ENCRYPTION KEY
+    | COLUMN MASTER KEY
+    | CONTRACT
+    | CREDENTIAL
+    | CRYPTOGRAPHIC PROVIDER
+    | DATABASE
+    | DATABASE AUDIT SPECIFICATION
+    | DATABASE ENCRYPTION KEY
+    | DATABASE EVENT SESSION
+    | DATABASE SCOPED CONFIGURATION
+    | DATABASE SCOPED CREDENTIAL
+    | DATABASE SCOPED RESOURCE GOVERNOR
```

Comment:
What do you think about simplifying by the following way? Here and in other places.

```suggestion
    | DATABASE
        ( AUDIT SPECIFICATION
        | ENCRYPTION KEY
        | EVENT SESSION
        | SCOPED ( CONFIGURATION
                 | CREDENTIAL
                 | RESOURCE GOVERNOR )
        )
```

It's good both for performance and clarity.

Reply:
Thanks for your suggestion! I have fixed it. Hope now is all right.